### PR TITLE
ENH: type np.unicode_ as np.str_

### PR DESCRIPTION
**Pull request by person142**
The two are aliases; i.e.

```
>>> np.unicode_ is np.str_
True
```

so make them aliases on the typing level too.

### CODE REVIEW

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -1527,6 +1526,8 @@ class str_(character, str):
         self, __value: bytes, encoding: str = ..., errors: str = ...
     ) -> None: ...
 
+unicode_ = str_
```
Comment (BvB93) [inline numpy/__init__.pyi:1529]:
```suggestion
unicode_ = str0 = Str0 = str_
```
`str0` and `Str0` are two other aliases:
``` python
In [1]: import numpy as np

In [2]: {k: getattr(np, k) for k in np.__all__ if getattr(np, k) is np.str_}
Out[2]: 
{'str0': numpy.str_,
 'Str0': numpy.str_,
 'str_': numpy.str_,
 'unicode_': numpy.str_}
```

Comment (person142) [reply numpy/__init__.pyi]: Added `str0`

**Review by BvB93 — APPROVED** [review]
LGTM; CI failure is unrelated (see https://github.com/numpy/numpy/pull/17499).

**Comment by charris** [comment]
Thanks @person142 .

**Pull request merged**